python/az/aro/azext_aro/custom.py

```diff
@@ -452,35 +453,57 @@ def aro_update(cmd,
                refresh_cluster_credentials=False,
                client_id=None,
                client_secret=None,
+               platform_workload_identities=None,
                load_balancer_managed_outbound_ip_count=None,
                no_wait=False):
     # if we can't read cluster spec, we will not be able to do much. Fail.
     oc = client.open_shift_clusters.get(resource_group_name, resource_name)
 
-    ocUpdate = openshiftcluster.OpenShiftClusterUpdate()
+    oc_update = openshiftcluster.OpenShiftClusterUpdate()
+
+    if platform_workload_identities is not None and oc.service_principal_profile is not None:
+        raise InvalidArgumentValueError(
+            "Cannot assign platform workload identities to a cluster with service principal"
+        )
+
+    if oc.service_principal_profile is not None:
+        client_id, client_secret = cluster_application_update(cmd.cli_ctx, oc, client_id, client_secret, refresh_cluster_credentials)  # pylint: disable=line-too-long
 
-    client_id, client_secret = cluster_application_update(cmd.cli_ctx, oc, client_id, client_secret, refresh_cluster_credentials)  # pylint: disable=line-too-long
+        if client_id is not None or client_secret is not None:
+            # construct update payload
+            oc_update.service_principal_profile = openshiftcluster.ServicePrincipalProfile()
 
-    if client_id is not None or client_secret is not None:
-        # construct update payload
-        ocUpdate.service_principal_profile = openshiftcluster.ServicePrincipalProfile()
+            if client_secret is not None:
+                oc_update.service_principal_profile.client_secret = client_secret
 
-        if client_secret is not None:
-            ocUpdate.service_principal_profile.client_secret = client_secret
+            if client_id is not None:
+                oc_update.service_principal_profile.client_id = client_id
 
-        if client_id is not None:
-            ocUpdate.service_principal_profile.client_id = client_id
+    if platform_workload_identities is not None:
+        pwis = {}
+        for i in oc.platform_workload_identity_profile.platform_workload_identities:
+            pwis[i.operator_name] = openshiftcluster.PlatformWorkloadIdentity(
+                operator_name=i.operator_name,
+                resource_id=i.resource_id
+            )
+
+        for i in platform_workload_identities:
+            pwis[i.operator_name] = i
+
+        oc_update.platform_workload_identity_profile = openshiftcluster.PlatformWorkloadIdentityProfile(
+            platform_workload_identities=list(pwis.values())
+        )
```

Comment:
It looks like if a user were to provide multiple PWIs with the same operator name in their request, this usage of a map with the operator names as keys would result in one of those PWIs being used and the others being ignored. This could result in some subtle issues, since the CLI would send a valid request to the RP that may be different from what the user intended. It's unlikely but possible.

What do you think of addressing this by checking for duplicate operator names in `validate_platform_workload_identities` and adding a corresponding test case?

Reply:
I like that idea and agree we should prevent that CLI-side (though our API should also reject requests with multiple identities for the same platform workload). I'll make that change.

Reply:
pushed up an implementation for this, lmk what you think

Reply:
Looks good for the most part, but I left one comment/change request in a separate discussion thread.